File: mvvm/src/main/java/com/hhh/mvvm/recycler/RecyclerViewHolder.java
package com.hhh.mvvm.recycler;

import android.view.View;

import com.hhh.mvvm.base.BaseActivity;
import com.hhh.mvvm.listener.OnImageShowListener;

import androidx.annotation.CallSuper;
import androidx.annotation.IdRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

public abstract class RecyclerViewHolder<MODEL> extends RecyclerView.ViewHolder {

  protected MODEL mMODEL;

  public RecyclerViewHolder(@NonNull View itemView) {
    super(itemView);
  }

  @CallSuper
  public void bind(@NonNull MODEL model) {
    mMODEL = model;
  }

  @NonNull
  public MODEL getModel() {
    return mMODEL;
  }

  @Nullable
  public OnImageShowListener getImageShowListener() {
    return null;
  }

  @Nullable
  public final <T extends View> T findViewById(@IdRes int id) {
    return itemView.findViewById(id);
  }

  @NonNull
  public BaseActivity getActivity() {
    return (BaseActivity) itemView.getContext();
  }
}

File: core/src/main/java/com/nb/core/RecyclerViewHolder.kt
package com.nb.core

import android.view.View
import androidx.annotation.CallSuper
import androidx.lifecycle.DefaultLifecycleObserver
import androidx.recyclerview.widget.RecyclerView

open class RecyclerViewHolder<MODEL : DiffCallback<MODEL>>(view: View) :
    RecyclerView.ViewHolder(view), DefaultLifecycleObserver, RecyclerFragmentInjection {

    protected lateinit var model: MODEL
    protected lateinit var recyclerFragment: RecyclerFragment<MODEL, *>

    override fun inject(fragment: RecyclerFragment<*, *>) {
        recyclerFragment = fragment as RecyclerFragment<MODEL, *>
    }

    @CallSuper
    open fun bind(model: MODEL) {
        this.model = model
    }
}